crates/red_knot_python_semantic/src/types.rs

```diff
@@ -819,6 +848,11 @@ impl<'db> Type<'db> {
                 // are both of type Literal[345], for example.
                 false
             }
+            Type::Type(..) => {
+                // TODO once we have support for final classes, we can return `true` for some
+                // cases: type[C] is a singleton if C is final.
```

Comment:
And in this case `type[C]` is equivalent to `Literal[C]` -- not sure if that's worth a TODO comment in `is_equivalent_to`, or a mention in this comment?

Reply:
> a TODO comment in `is_equivalent_to`

:+1: